test/fw/ptl/utils/pbs_logutils.py

```diff
@@ -296,7 +296,7 @@ def open_log(self, log, hostname=None, sudo=False):
         try:
             if hostname is None or self.du.is_localhost(hostname):
                 if sudo:
-                    cmd = copy.copy(self.sudo_cmd) + ['cat', log]
+                    cmd = copy.copy(self.du.sudo_cmd) + ['cat', log]
```

Comment:
This might be better for another ticket, but if you have access to du inside of log utils, why not use self.du.cat instead of directly calling Popen()?

Reply:
@bhroam open_log returns File like instance which caller can iterate to get lines one by one. Where DshUtils.cat returns contents of the file as a list of strings instead File like object.

But saying that we can do one thing here as the list is also iterable we can either return a list of lines or if we still want File like object after using self.du.cat then we can make use of StringIO to get File like object from string.

Reply:
> From what I know, File objects are better than lists because they take less memory.

True


> I don't see the File object we return helping us any more than returning a list. Please correct me if I am wrong.

True